pkg/objects/source.go

```diff
@@ -373,6 +373,9 @@ func (u URLSource) getFileContents(path string, cache map[string]FileSpec) (cont
 	if err != nil {
 		return nil, uri, &ListEntriesError{uri, "GET failed: " + err.Error()}
 	}
+	if resp.StatusCode >= 400 {
+		return nil, uri, &ListEntriesError{uri, "GET failed: " + resp.Status}
```

Comment:
I'd rather not rely on `resp.Status` since the server is free to give any random message there. (`Status` is for human consumption, e.g. in logs, whereas `StatusCode` is for consumption by machines.)

Also, this does not match the error format that you're looking for in line 250 of `debian.go`. I suggest the following to fix both issues:

```suggestion
		return nil, uri, &ListEntriesError{uri, fmt.Sprintf("GET returned status %d", resp.StatusCode)}
```